components/esp32/panic.c

```diff
@@ -212,7 +212,10 @@ void panicHandler(XtExcFrame *frame)
                     //debugcause if the cause is watchdog 1 and clearing it if it's watchdog 0.
                     if (debugRsn&(1<<8)) {
 #if CONFIG_FREERTOS_WATCHPOINT_END_OF_STACK
-                        panicPutStr("Stack canary watchpoint triggered ");
+                        const char *name = pcTaskGetTaskName(xTaskGetCurrentTaskHandleForCPU(core_id));
```

Comment:
This pointer will likely be in cached region of memory. Would it make sense to sanity-check it before trying to print it? I.e. something like
```
if (spi_flash_cache_enabled() || !((uint32_t) name >= SOC_DROM_LOW && (uint32_t) name < SOC_DROM_HIGH)) 
```